src/System.Net.Http/src/uap/System/Net/HttpHandlerToFilter.cs

```diff
@@ -26,65 +26,207 @@ namespace System.Net.Http
+                {
+                    if (redirects > 0)
+                    {
+                        rtResponse = await _filterWithNoCredentials.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        rtResponse = await _filter.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+                    }
+                }
+                catch (TaskCanceledException)
+                {
+                    throw;
+                }
+                catch (Exception)
+                {
+                    object info;
+                    if (rtRequest.Properties.TryGetValue(SavedExceptionDispatchInfoLookupKey, out info))
+                    {
+                        ((ExceptionDispatchInfo)info).Throw();
+                    }
+
+                    throw;
+                }
+
+                response = ConvertResponse(rtResponse);
+
+                ProcessResponseCookies(response, request.RequestUri);
+
+                if (!_handler.AllowAutoRedirect)
+                {
+                    break;
+                }
+
+                if (response.StatusCode != HttpStatusCode.MultipleChoices &&
+                    response.StatusCode != HttpStatusCode.MovedPermanently &&
+                    response.StatusCode != HttpStatusCode.Redirect &&
+                    response.StatusCode != HttpStatusCode.RedirectMethod &&
+                    response.StatusCode != HttpStatusCode.RedirectKeepVerb)
+                {
+                    break;
+                }
+
+                redirects++;
+                if (redirects > _handler.MaxAutomaticRedirections)
+                {
+                    break;
+                }
+
+                Uri redirectUri = response.Headers.Location;
+                if (redirectUri == null)
+                {
+                    break;
+                }
+
+                if (!redirectUri.IsAbsoluteUri)
+                {
+                    redirectUri = new Uri(string.Format("{0}://{1}:{2}{3}",
+                        request.RequestUri.Scheme,
+                        request.RequestUri.Host,
+                        request.RequestUri.Port,
+                        redirectUri.OriginalString));
+                }
+
+                if (redirectUri.Scheme != Uri.UriSchemeHttp &&
+                    redirectUri.Scheme != Uri.UriSchemeHttps)
+                {
+                    break;
+                }
+
+                if (request.RequestUri.Scheme == Uri.UriSchemeHttps &&
+                    redirectUri.Scheme == Uri.UriSchemeHttp)
+                {
+                    break;
+                }
+
+                if (response.StatusCode != HttpStatusCode.RedirectKeepVerb &&
+                    requestHttpMethod == HttpMethod.Post)
+                {
+                    requestHttpMethod = HttpMethod.Get;
+                    skipRequestContentIfPresent = true;
```

Comment:
Also, it would be nice to have a comment referencing the Post/Redirect/Get pattern since it's not obvious why a Post becomes a Get.